manifests/public_key.pp

```diff
@@ -1,12 +1,27 @@
-class bootstrap::public_key ( 
-  $admin_user = $bootstrap::params::admin_user,
-  $ec2_lock_passwd = $bootstrap::params::ec2_lock_passwd
+class bootstrap::public_key (
+  String           $admin_user      = $bootstrap::params::admin_user,
+  Boolean          $ec2_lock_passwd = $bootstrap::params::ec2_lock_passwd,
+  String           $ssh_key         = $bootstrap::params::admin_ssh_key,
+  Optional[String] $additional_key  = undef,
 ) inherits bootstrap::params {
+
   ssh_authorized_key { 'instructor':
     user => $admin_user,
     type => 'ssh-rsa',
-    key  => 'AAAAB3NzaC1yc2EAAAADAQABAAABAQCv+89n9QU8kPxhsoRAHt5CZMy4cyA7vQ8k60wZmyUHnmFtXH8Ipusj9QyV8GsL/F64ci3bS0eHlk5XlAFACsLy4Gai6CQ2NAjkZRT7qG5BtoGr4aGXhxFCUZ2hgo2j18uU6GKl2akRPsYMsRgNrRglidENLY4WUCzacgr/1Cw8eRN95Mo9b4tgSLTCyd/783CBdeM8qU9Go7xq0nVRGNgsFIqf1SY+9k5dpSbL0M5lvDzl35IC1tRIQE/ZbafJok8g1XmhwKokWpGBRk9/GneWz3jDvJRZPjYHzQs+gSHNkl2F6i55EIlQqbfm/7lvtB73Pd2vT+i1AIgrx90YB4Yd',
+    key  => $ssh_key,
   }
+
+  if $additional_key {
```

Comment:
@binford2k Will this throw that "unknown variable" warning if the value is `undef`? Possibly replace with `defined('$additional_key')`?

Reply:
No, `undef` is fine. That warning is only thrown when a variable doesn't exist at all.

Reply:
Ok, I thought it was also if it was assigned `undef`. I prefer to be able to say `if $var { ... }`, rather than use `defined`.